Add statement on transaction relay (#1139)
bitcoin-core/bitcoincore.org: _posts/en/posts/2025-06-03-relay-statement.md

---
title: Bitcoin Core development and transaction relay policy

> permalink: /en/2025/06/03/relay-statement/

Reviewer: nit: could bump to the (assumed) merging/publishing date and rename file accordingly
```suggestion
permalink: /en/2025/06/05/relay-statement/
```

cheap. However this can only be effective while both users and miners are satisfied with whatever
alternatives exist. When that is no longer the case, and an economically viable use case develops
that would conflict with policy rules, users and miners can directly collaborate to avoid any
external attempt to impose restrictions on their activities. In fact, the ability to do precisely
that is an important aspect of Bitcoin's censorship resistance, and other node software with
preferential peering has also shown that circumventing filters of the vast majority of the nodes
is relatively easy. Given that, we believe it is better for Bitcoin node software to aim to have a
realistic idea of what will end up in the next block, rather than attempting to intervene between
consenting users and miners in order to discourage non-monetary activity that is largely harmless
at a technical level. This is not endorsing or condoning non-financial data usage, but accepting
that as a censorship-resistant system, Bitcoin can and will be used for use cases not everyone
agrees on.

While we recognize that this view isn't held universally by all users and developers, it is our

> sincere belief that it is in the best interest of Bitcoin and its users, and hope our users agree.

Reviewer 2: ```suggestion
sincere belief that it is in the best interest of Bitcoin and its users, and we hope our users agree.
```

---
title: Bitcoin Core development and transaction relay policy
permalink: /en/2025/06/03/relay-statement/
lang: en
id: en-relay-statement
name: relay-statement
type: posts
layout: post
version: 1
---

We'd like to share our view on the relationship between Bitcoin Core development and transaction relay
policy on the network.

Bitcoin is a network that is defined by its users, who have ultimate freedom in choosing what
software they use (fully-validating or not) and implementing whatever policies they desire. Bitcoin
Core contributors are not in a position to mandate what those are. One way this is reflected is by
our long-running practice of avoiding auto-updating in the software. This means that no entity can
unilaterally push out changes to Bitcoin Core users: changes must be made by users choosing to
adopt new software releases themselves, or if they so desire, different software. Being free to run
any software is the network's primary safeguard against coercion.

As Bitcoin Core developers we also consider it our responsibility to make our software work as
efficiently and reliably as possible for its purpose, namely validating and relaying blocks and

> transactions in the Bitcoin peer-to-peer network, so that it succeeds as a decentralized digital

Reviewer 3: "so that Bitcoin succeeds" might be clearer

Knowingly refusing to relay transactions that miners would include in blocks anyway forces users into
alternate communication channels, undermining the above goals.

It is the case that transaction acceptance rules have been used effectively in the past to
discourage the development of use cases that used block space inefficiently while doing so was very
cheap. However this can only be effective while both users and miners are satisfied with whatever
alternatives exist. When that is no longer the case, and an economically viable use case develops
that would conflict with policy rules, users and miners can directly collaborate to avoid any
external attempt to impose restrictions on their activities. In fact, the ability to do precisely
that is an important aspect of Bitcoin's censorship resistance, and other node software with
preferential peering has also shown that circumventing filters of the vast majority of the nodes
is relatively easy. Given that, we believe it is better for Bitcoin node software to aim to have a
realistic idea of what will end up in the next block, rather than attempting to intervene between

> consenting users and miners in order to discourage non-monetary activity that is largely harmless

Reviewer 4: Ah, I see. A more verbose alternative that clarifies this:

```suggestion
a user that wants to send a transaction and miners that are willing to mine it in order to discourage non-monetary activity that is largely harmless
```

No strong view, up to your preference, but I think this might avoid some misunderstandings.

---
title: Bitcoin Core development and transaction relay policy
permalink: /en/2025/06/03/relay-statement/
lang: en
id: en-relay-statement
name: relay-statement
type: posts
layout: post
version: 1
---

We'd like to share our view on the relationship between Bitcoin Core development and transaction relay

> policy on the network.

Reviewer 5: From @ryanofsky's comment, perhaps it would be useful to add some context for why we are making this statement? I can see how the first paragraph can otherwise be interpreted as "you should go somewhere else if you disagree" instead of "hey, we can't actually make the Bitcoin-wide changes people are requesting"

```suggestion
policy on the network. This statement was written in response to feedback that Bitcoin Core's transaction relay policy should be used to prevent transactions with embedded data from being mined.
```

Reviewer 6: I'm worried that the above addition might be construed as straw-manning the filterers position to simply be conflating policy and consensus. Might be worth having something like:

```suggestion
policy on the network. This statement was written in response to feedback that Bitcoin Core's transaction relay policy should be left alone/narrowed in an attempt to discourage transactions with embedded data from being mined.
```

But also okay without the addition.

preferential peering has also shown that circumventing filters of the vast majority of the nodes
is relatively easy. Given that, we believe it is better for Bitcoin node software to aim to have a
realistic idea of what will end up in the next block, rather than attempting to intervene between
consenting transaction creators and miners in order to discourage activity that is largely harmless
at a technical level.

**This is not endorsing or condoning non-financial data usage, but accepting
that as a censorship-resistant system, Bitcoin can and will be used for use cases not everyone
agrees on.**

While we recognize that this view isn't held universally by all users and developers, it is our
sincere belief that it is in the best interest of Bitcoin and its users, and we hope our users agree.
We will continue to apply our best judgment as developers in aligning transaction acceptance rules
with Bitcoin's long-term health and miners' rational self-interest, including specific
technical reasons such as upgrade safety, efficient block building, and node DoS attacks.

Signed,

(List of contributors who support this letter)

* Andrew Toth
* Antoine Poinsot

> * Antony Towns

Reviewer 7: ```suggestion
* Anthony Towns
```

* Pol Espinasa
* Sergi Delgado
* Will Clark

> * w0xlt

Reviewer 8: nzengi
